designs/DotNetSignCommands.md

```diff
@@ -0,0 +1,141 @@
+customers.
+
+## Who are the customers
+
+This feature is for `dotnet.exe` users who want to take advantage of more
+package signing-related features of nuget without needing to install
+`nuget.exe`.
+
+## Goals
+
+Design and implement equivalent commands for the following `nuget.exe` commands:
+
+- `nuget.exe sign`
+- `nuget.exe verify`
+- `nuget.exe trusted-signers`
+
+These commands should have `dotnet.exe`-compatible syntax and ergonomics, and
+feel like proper `dotnet.exe` commands.
+
+## Non-Goals
+
+Adding additional signing-related commands, or significantly
+changing/improving the behavior of the commands as currently designed.
+
+## Solution
+
+The following commands will be implemented in the `dotnet.exe` CLI.
+
+Options marked with `*` indicate a notable change in behavior/naming.
+
+### `dotnet nuget sign`
+
+```
+Signs a NuGet package at <package-path> with the specified certificate.
+
+USAGE:
+    dotnet nuget sign [OPTIONS] <package-path>
+
+OPTIONS:
+    -o, --output* <output-directory>    Directory where the signed package should be saved. By default, the original package is overwritten by the signed package.
+        --certificate-path              File path to the certificate to be used while signing the package.
+        --certificate-store-name        Name of the X.509 certificate store. (...)
+        --certificate-store-location    Name of the X.509 certificate store to use to search for the certificate.
+        --certificate-subject-name      Subject name of the certificate used to search a local certificate store for the certificate.
+        --certificate-fingerprint       SHA-1 fingerprint of the certificate used to search a local certificate store for the certificate.
+        --certificate-password          Password for the certificate, if needed.
+        --hash-algorithm                Hash algorithm to be used to sign the package. Defaults to SHA256.
+        --timestamper                   URL to an RFC 3161 timestamping server.
+        --timestamp-hash-algorithm      Hash algorithm to be used by the RFC 3161 timestamping server. Defaults to SHA256.
+        --overwrite                     Switch to indicate if the current signature should be overwritten.
+    -h, --help                          Prints help information.
+    -v, --verbosity* <LEVEL>            Set the MSBuild verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+        --interactive*                  Allows command to stop and wait for user input or action.
+        --configfile                    The NuGet configuration file. If specified, only the settings from this file will be used. If not specified, the hierarchy of configuration files from the current directory will be used.
+
+EXAMPLES:
+
+    dotnet nuget sign MyPackage.nupkg --timestamper https://foo.bar
+
+    dotnet nuget sign ../MyPackage.nupkg --timestamper https://foo.bar --output-directory ../signed
```

Comment:
This example has the options in the end. Seems need to be changed?
And pls also change` --output-directory` to `--output`

Reply:
Yeah, they can be in either position. That's pretty normal. I'll reorder them to avoid confusion, though.

And thanks! Nice catch on `--output-directory` 👍🏻